martin/src/bin/martin-cp.rs

```diff
@@ -566,12 +588,97 @@ async fn main() {
 
 #[cfg(test)]
 mod tests {
+    use super::*;
+    use async_trait::async_trait;
+    use insta::assert_yaml_snapshot;
+    use martin::TileSources;
+    use martin_core::tiles::{MartinCoreResult, Source, UrlQuery};
+    use martin_tile_utils::{Encoding, Format};
+    use rstest::{fixture, rstest};
     use std::str::FromStr;
+    use tilejson::{TileJSON, tilejson};
 
-    use insta::assert_yaml_snapshot;
-    use rstest::rstest;
+    #[derive(Debug, Clone)]
+    pub struct MockSource {
+        pub id: &'static str,
+        pub tj: TileJSON,
+        pub data: TileData,
+    }
```

Comment:
It looks like there's a couple ways mocks and fixtures are organized in the project. Any thoughts on these for bins?

Reply:
Bins? Not sure I follow what you mean.

Current tests you added look very good if this is what you are asking 👍

Reply:
Bins? Not sure I follow what you mean.

Current tests you added look very good if this is what you are asking 👍